pkg/server/leases/gc_controller_test.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package leases
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	clocktesting "k8s.io/utils/clock/testing"
+
+	proxytesting "sigs.k8s.io/apiserver-network-proxy/pkg/testing"
+)
+
+func TestGarbageCollectionController(t *testing.T) {
+	testCases := []struct {
+		name           string
+		template       proxytesting.LeaseTemplate
+		selector       string
+		expectDeletion bool
+	}{
+		{
```

Comment:
nit: mixing time.Minute and DurationSecs makes me have to do mental math, we could use time.Second instead